card/mod/standard/set/all/links.rb

```diff
@@ -80,79 +71,126 @@ def view_link text, view, opts={}
   end
 
   def related_link name_or_card, opts={}
-    name =
-      case name_or_card
-      when Symbol then Card.fetch(name_or_card, skip_modules: true).cardname
-      when Card   then name_or_card.cardname
-      else             name_or_card
-      end
+    name = Card::Name.cardish name_or_card
     opts[:path_opts] ||= { view: :related }
     opts[:path_opts][:related] = { name: "+#{name}" }
     opts[:path_opts][:related].merge! opts[:related_opts] if opts[:related_opts]
     view_link(opts[:text] || name, :related, opts)
   end
 
-  def path opts={}
-    if opts[:action] == :new && opts[:type] &&
-       !(opts[:name] || opts[:card] || opts[:id])
-      opts.delete(:action)
-      base = "new/#{opts.delete(:type)}"
-    else
-      name = opts.delete(:name) || card.name
-      base = opts[:action] ? "card/#{opts.delete :action}/" : ""
+  # @param opts [Hash]
+  # @option opts [Symbol] :action card action (:create, :update, :delete)
+  # @option opts [Integer, String] :id
+  # @option opts [String, Card::Name] :name
+  # @option opts [String] :type
```

Comment:
I can see altering the behavior here, but it's not immediately clear what we want.  Here's what currently happens:
1. if action is :new (the only not-a-real-action action supported), we use it to produce /new/Cardtype
2. otherwise if opts[:type] is a known card, we move it to params[:card][:type]

If we wanted to support symbols or Integers, would we want to turn them into cardname based paths or would we want to use our stringy versions and support things like new/:type_search?